src/Models/Cart.php

```diff
@@ -350,6 +363,59 @@ public function scopeActive(Builder $query): Builder
             });
     }
 
+    /******************************
+     * CartInterface Implementation
+     ******************************/
+
+    private static function activeDeductions(): Collection
+    {
+        return collect(static::$deductionTypes)
+            ->filter(function (string $type) {
+                return app()->has($type);
+            })
+            ->map(function (string $type) {
+                return app($type);
+            });
```

Comment:
This is code that determines which CartDeduction interfaces are actually registered and returns a collection of the instantiated objects.